code/software/libs/src/sst_flash/sst.c

```diff
@@ -0,0 +1,139 @@
+/*
+ *------------------------------------------------------------
+ *                                  ___ ___ _   
+ *  ___ ___ ___ ___ ___       _____|  _| . | |_ 
+ * |  _| . |_ -|  _| . |     |     | . | . | '_|
+ * |_| |___|___|___|___|_____|_|_|_|___|___|_,_| 
+ *                     |_____|         tech demo
+ * ------------------------------------------------------------
+ * Copyright (c)2021 Ross Bamford
+ * See top-level LICENSE.md for licence information.
+ *
+ * SST39SF0x0 Flash EEPROM interface library
+ * ------------------------------------------------------------
+ */
+
+#include "sst_flash.h"
+
+#define SST_UNLOCK_A1       ((0x00005555<<1))
+#define SST_UNLOCK_D1       0xAA
+#define SST_UNLOCK_A2       ((0x00002AAA<<1))
+#define SST_UNLOCK_D2       0x55
+
+#define SST_SUBCMD_SECERASE 0x30
+#define SST_SUBCMD_CHPERASE 0x10
+
+static void write(uint32_t addr, uint8_t data) {
+    uint8_t *p = (uint8_t*)addr;
```

Comment:
It's a minor thing, and I'm not sure it's needed, but I'd consider making all the `uint8_t *p` be `volatile`, just to make sure that the compiler doesn't have the chance to inline and optimize-out any writes at some point in the future.

Reply:
See reply to your other comment, but long story short this is totally necessary, good spot!

With this being `volatile` and the `wait`s removed, I'm now getting this:

```
Kernel received okay; Starting...


rosco_m68k Flash EEPROM identifier / write POC
**********************************************

EVEN ROM is SST39SF040  [Manufacturer SST]
ODD  ROM is SST39SF040  [Manufacturer SST]

Config area is 1 sector per ROM (2 total)
  Sector num : 0x7f
  Sector size: 4096 bytes
  Total size : 8192 bytes
  Base addr  : 0x00efe000

Old data: 0x55aa55aa 0x55aa55aa 0x55aa55aa...
New data: 0xaa55aa55 0xaa55aa55 0xaa55aa55...
```